### PR TITLE
quic: simplifications, cleanups, and fixes

### PR DESCRIPTION
**Pull request by jasnell**
Multiple simplifications, cleanups, etc

##### Checklist

- [ ] `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/doc/guides/contributing/pull-requests.md#commit-message-guidelines)

### CODE REVIEW

File: doc/api/quic.md [hunk 1]
```diff
@@ -305,6 +305,20 @@ added: REPLACEME
...
 True if the TLS handshake has completed.
 
+### quicsession.maxStreams
+<!-- YAML
+added: REPLACEME
+-->
+
+* Type: {Object}
+  * `uni` {number} The maximum number of unidirectional streams.
+  * `bidi` {numbe} The maximum number of bidirectional streams.
```
Comment (danbev) [inline doc/api/quic.md]: Minor typo: `numbe`

File: lib/internal/quic/core.js [hunk 2]
```diff
@@ -581,6 +570,9 @@ function onNewListener(event) {
     case 'clientHello':
       this[kHandle].state[IDX_QUIC_SESSION_STATE_CLIENT_HELLO_ENABLED] = 1;
       break;
+    case 'pathValidation':
+        this[kHandle].state[IDX_QUIC_SESSION_STATE_PATH_VALIDATED_ENABLED] = 1;
```
Comment (danbev) [inline lib/internal/quic/core.js]: Indenting is a little off on this line and the next.

File: doc/api/quic.md [hunk 3]
```diff
@@ -252,8 +229,8 @@ added: REPLACEME
 * `error` {any}
 
 Destroys the `QuicSession` immediately causing the `close` event to be emitted.
-If `error` is not `undefined`, the `error` event will be emitted following the
-`close` event.
+If `error` is not `undefined`, the `error` event will be emitted immediately before
```
Comment (danbev) [inline doc/api/quic.md]: This is giving me a warning regarding the length of the line:
```console
232:84  warning  Line must be at most 80 characters  maximum-line-length  remark-lin
```

**Comment by danbev** [comment]
I noticed that there is also an lint error unrelated to this PR:
```console
 1203:17  error  'ERR_QUIC_SOCKETDESTROYED' is not defined     no-undef
```
Which should be [ERR_QUICSOCKETDESTROYED](https://github.com/nodejs/quic/blob/e9e4488228013bc1c83fa98bd3d65f587cc64fb2/lib/internal/errors.js#L1120). I can raise a PR but if thinking that you might be hitting this too.

**Comment by jasnell** [comment]
I'll fix that lint issue in a bit, but also, feel free to push a commit to this PR if you happen to beat me to it :-)

**Comment by danbev** [comment]
> I'll fix that lint issue in a bit, but also, feel free to push a commit to this PR if you happen to beat me to it :-)

:) No problems, added one for it (https://github.com/nodejs/quic/pull/101)

**Comment by jasnell** [comment]
Landed!

**Pull request closed without merging**